src/repositories/ImmutableRepository.ts

```diff
@@ -0,0 +1,124 @@
+    super(entity, manager);
+    this.primaryKey = primaryKey;
+  }
+
+  /********************************** Public Interfaces **********************************/
+  /**
+   * Adds a new entity to the repository using the provided body.
+   *
+   * @param body The body containing the entity data.
+   * @returns A promise that resolves to the newly created entity.
+   */
+  public abstract addOne(body: CreateBody<T>): Promise<T>;
+
+  /**
+   * Retrieves an entity with the specified primary key.
+   *
+   * @param primaryKey The primary key of the entity to retrieve.
+   * @returns A promise that resolves to the entity if found, or null if not found.
+   */
+  public async getOne(primaryKey: string): Promise<T | null> {
+    return await this.findOne(primaryKey);
+  }
+
+  /********************************** Subclass Methods **********************************/
+  /**
+   * Throws an IllegalArgumentException if the given primary key is undefined or an empty array.
+   *
+   * @param primaryKey The primary key to check.
+   */
+  protected throwsIfPKIsUndefined(primaryKey: string | string[]) {
+    if (!primaryKey) {
+      throw new IllegalArgumentException(`The given ${String(this.primaryKey)} is undefined!`);
+    }
+
+    if (Array.isArray(primaryKey) && primaryKey.length === 0) {
+      throw new IllegalArgumentException(`The given array of ${String(this.primaryKey)} is empty!`);
+    }
+  }
+
+  /**
+   * Finds a single entity with the specified primary key and optional options.
+   *
+   * @param primaryKey The primary key of the entity to find.
+   * @param options Optional options for finding the entity, such as relations, join, withDeleted, select, and order.
+   * @throws IllegalArgumentException if the given PK is undefined or empty.
+   * @returns A promise that resolves to the found entity, or null if not found.
+   */
+  protected async findOne(
+    primaryKey: string,
+    options: Pick<
+      FindOneOptions<T>,
+      'relations' | 'join' | 'withDeleted' | 'select' | 'order'
+    > = {},
+  ): Promise<T | null> {
+    this.throwsIfPKIsUndefined(primaryKey);
+
+    return await this.repository.findOne({
+      where: {
+        [this.primaryKey]: primaryKey,
+      } as FindOptionsWhere<T>,
+      ...options,
+    });
+  }
+
+  /**
+   * Inserts a new entity into the repository.
+   *
+   * @param entity The entity data to insert.
+   * @param relations Optional relations to load with the inserted entity.
+   * @throws EntryNotFoundAfterInsertException if the inserted entity is not found.
+   * @returns A promise that resolves to the inserted entity.
+   */
+  protected async insert(entity: DeepPartial<T>, relations?: FindOptionsRelations<T>): Promise<T> {
+    try {
+      const insertResult = await this.repository.insert(entity as QueryDeepPartialEntity<T>);
+      const insertedEntity = await this.findOne(insertResult.identifiers[0].id, { relations });
```

Comment:
I'm probably a bit anxious, but I'm worried about : What if insertResult.identifiers.length is equals to 0 ?
It would throw an OutOfBand exception, that is catched and rethrowed as an InsertionException.

This behavior seems okay to me !